domain/info.go

```diff
@@ -0,0 +1,220 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package domain
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/clientv3/concurrency"
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/config"
+	"github.com/pingcap/tidb/ddl"
+	"github.com/pingcap/tidb/mysql"
+	"github.com/pingcap/tidb/owner"
+	"github.com/pingcap/tidb/util/hack"
+	"github.com/pingcap/tidb/util/printer"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+)
+
+const (
+	// ServerInformationPath store server information such as IP, port and so on.
+	ServerInformationPath = "/tidb/server/info"
+	// keyOpDefaultRetryCnt is the default retry count for etcd store.
+	keyOpDefaultRetryCnt = 2
+	// keyOpDefaultTimeout is the default time out for etcd store.
+	keyOpDefaultTimeout = 1 * time.Second
+)
+
+// InfoSessionTTL is the etcd session's TTL in seconds. It's exported for testing.
+var InfoSessionTTL = 10 * 60
+
+// InfoSyncer stores server info to etcd when when the tidb-server starts and delete when tidb-server shuts down.
+type InfoSyncer struct {
+	etcdCli        *clientv3.Client
+	info           *ServerInfo
+	serverInfoPath string
+	session        *concurrency.Session
+}
+
+// ServerInfo is server static information.
+// It will not update when server running. So please only put static information in ServerInfo struct.
```

Comment:
s/update/be updated/ or change
when server running -> during tidb-server process lifetime